### PR TITLE
enable json functions pushdown.

### PR DESCRIPTION
**Pull request by hicqu**
enable json functions push down to tikv.

Now, it's waiting for JSON_REMOVE in TiKV.

All blocking works are done.

### CODE REVIEW

**Comment by tiancaiamao** [comment]
@AndreMouche

**Comment by hicqu** [comment]
@AndreMouche , PTAL, thanks.

**Comment by shenli** [comment]
LGTM
@AndreMouche PTAL

**Comment by ngaut** [comment]
LGTM

File: store/tikv/coprocessor.go [hunk 1]
```diff
@@ -70,7 +70,7 @@ func supportExpr(exprType tipb.ExprType) bool {
 	case tipb.ExprType_JsonType, tipb.ExprType_JsonExtract, tipb.ExprType_JsonUnquote, tipb.ExprType_JsonValid,
 		tipb.ExprType_JsonObject, tipb.ExprType_JsonArray, tipb.ExprType_JsonMerge, tipb.ExprType_JsonSet,
 		tipb.ExprType_JsonInsert, tipb.ExprType_JsonReplace, tipb.ExprType_JsonRemove, tipb.ExprType_JsonContains:
-		return false
+		return true
```
Comment (AndreMouche) [inline store/tikv/coprocessor.go]: It seems `tikv` haven't done the  implementation for ` tipb.ExprType_JsonValid`, `tipb.ExprType_JsonRemove`, `tipb.ExprType_JsonContains`

Comment (hicqu) [inline store/tikv/coprocessor.go]: json_valid and json_contains are not implemented in tidb either. So you only need to implement json_remove now.

File: store/tikv/coprocessor.go [hunk 2]
```diff
@@ -67,10 +67,10 @@ func supportExpr(exprType tipb.ExprType) bool {
 		return true
 	case tipb.ExprType_Count, tipb.ExprType_First, tipb.ExprType_Max, tipb.ExprType_Min, tipb.ExprType_Sum, tipb.ExprType_Avg:
 		return true
-	case tipb.ExprType_JsonType, tipb.ExprType_JsonExtract, tipb.ExprType_JsonUnquote, tipb.ExprType_JsonValid,
-		tipb.ExprType_JsonObject, tipb.ExprType_JsonArray, tipb.ExprType_JsonMerge, tipb.ExprType_JsonSet,
-		tipb.ExprType_JsonInsert, tipb.ExprType_JsonReplace, tipb.ExprType_JsonRemove, tipb.ExprType_JsonContains:
-		return false
+	case tipb.ExprType_JsonType, tipb.ExprType_JsonExtract, tipb.ExprType_JsonUnquote,
+		tipb.ExprType_JsonObject, tipb.ExprType_JsonArray, tipb.ExprType_JsonMerge,
+		tipb.ExprType_JsonSet, tipb.ExprType_JsonInsert, tipb.ExprType_JsonReplace, tipb.ExprType_JsonRemove:
```
Comment (shenli) [inline store/tikv/coprocessor.go]: ExprType_JsonRemove is not implemented yet.

Comment (hicqu) [inline store/tikv/coprocessor.go]: Yes, so we need wait that.

Comment (AndreMouche) [inline store/tikv/coprocessor.go]: `Json_remove` in `tikv` is on the way to merge, may be you could merge this PR later than the related PR in tikv(https://github.com/pingcap/tikv/pull/2091) @hicqu

**Comment by tiancaiamao** [comment]
Is this ready now? @hicqu @AndreMouche

**Comment by AndreMouche** [comment]
yes @tiancaiamao

**Comment by AndreMouche** [comment]
LGTM

**Pull request merged**